pkg/api/util/conditions.go

```diff
@@ -147,16 +148,20 @@ func SetCertificateCondition(crt *cmapi.Certificate, conditionType cmapi.Certifi
 	nowTime := metav1.NewTime(Clock.Now())
 	newCondition.LastTransitionTime = &nowTime
 
+	// Set the condition generation to be that of the given Certificate spec
+	newCondition.ObservedGeneration = crt.Generation
+
 	// Search through existing conditions
 	for idx, cond := range crt.Status.Conditions {
 		// Skip unrelated conditions
 		if cond.Type != conditionType {
 			continue
 		}
 
-		// If this update doesn't contain a state transition, we don't update
-		// the conditions LastTransitionTime to Now()
-		if cond.Status == status {
+		// If this update doesn't contain a state transition, and this condition is
+		// for the same Certificate generation, we don't update the conditions
+		// LastTransitionTime to Now()
+		if cond.Status == status && cond.ObservedGeneration == crt.Generation {
```

Comment:
The generation changing should _not_ be considered a transition, so I don't think we should have this extra clause.